pkg/roachpb/metadata.go

```diff
@@ -457,6 +467,55 @@ func (r ReplicaDescriptor) GetType() ReplicaType {
 	return *r.Type
 }
 
+// Equal compares two descriptors for equality. This was copied over from the
+// gogoproto generated version in order to handle Type == nil equally to
+// Type == 0. This is necessary because of a mixed-version bug, where 22.2 nodes
+// will always send Type == 0 in e.g. AdminChangeReplicas, but the 22.1 nodes
+// compare this against the range descriptor from KV for equality. See:
+// https://github.com/cockroachdb/cockroach/issues/94834
```

Comment:
Will is be possible to revert this code in some future version? Maybe note it too.

Reply:
This code doesn't exist in 22.2 or later, we're committing this to 22.1 only. In 22.2 and later the fields are non-nullable, so this doesn't apply.